sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs

```diff
@@ -264,7 +259,57 @@ public virtual PurchasePhoneNumbersOperation StartPurchasePhoneNumbers(string se
             scope.Start();
             try
             {
-                var originalResponse = InternalClient.StartPurchasePhoneNumbers(searchId, cancellationToken);
+                var originalResponse = InternalClient.StartPurchasePhoneNumbers(searchId, false, cancellationToken);
+                return new PurchasePhoneNumbersOperation(originalResponse);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Purchases phone numbers. </summary>
+        /// <remarks>
+        /// The agreement to not resell is a legal requirement in some countries in order to purchase phone numbers.
+        /// For more information on which countries require this agreement, please refer to this documentation:
+        /// https://learn.microsoft.com/azure/communication-services/concepts/numbers/sub-eligibility-number-capability
+        /// </remarks>
+        /// <param name="searchId"> The search id. </param>
+        /// <param name="agreeToNotResell"> The user-provided agreement to not resell the numbers purchased by this operation. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<PurchasePhoneNumbersOperation> StartPurchasePhoneNumbersAsync(string searchId, bool agreeToNotResell, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope($"{nameof(PhoneNumbersClient)}.{nameof(StartPurchasePhoneNumbers)}");
+            scope.Start();
+            try
+            {
+                var originalResponse = await InternalClient.StartPurchasePhoneNumbersAsync(searchId, agreeToNotResell, cancellationToken).ConfigureAwait(false);
+                return new PurchasePhoneNumbersOperation(originalResponse);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Purchases phone numbers. </summary>
+        /// <remarks>
+        /// The agreement to not resell is a legal requirement in some countries in order to purchase phone numbers.
+        /// For more information on which countries require this agreement, please refer to this documentation:
+        /// https://learn.microsoft.com/azure/communication-services/concepts/numbers/sub-eligibility-number-capability
+        /// </remarks>
+        /// <param name="searchId"> The search id. </param>
+        /// <param name="agreeToNotResell"> The user-provided agreement to not resell the numbers purchased by this operation. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual PurchasePhoneNumbersOperation StartPurchasePhoneNumbers(string searchId, bool agreeToNotResell, CancellationToken cancellationToken = default)
```

Comment:
Should this have a default value to false?

Reply:
In this case no because there already exists another version of the method without the `agreeToNotResell` argument. If I add a default value here, the compiler is unable to distinguish between calling this method with the default value or calling the old one. 

I am also unable to update the old method, since it would break backwards compatibility to add a new parameter.